### ISSUE TITLE
Puppeteer async doesn't work. It does pages one by one

### ISSUE THREAD

**Issue by 8m0** [hardkoded/puppeteer-sharp#2026]
### Description

So i simply trying to get values from multiple pages simultaneously. But it seems it doesn't work. Moreover, all tabs opened in order, gets lowered internet speed or maybe a page waits other page to done it's work, and actions in Network tab execute slower and slower. You can see it on the image below.
If number in braces more than 0, then on a page 0 were opened other pages to get values from. That's why it ends always with 0. 

I tried already to play with Args, but it doesn't affect anything reasonably.  **What am i missing?**

![Screenshot_1](https://user-images.githubusercontent.com/33127649/196232215-ba4c0c26-1b45-4f23-9dba-097e66075cc3.png)

### Code

```csharp
private async Task GetValuesFromBrowser()
{
        await using Browser browser = await Puppeteer.LaunchAsync(new LaunchOptions
        {
            Headless = true,
            Args = new string[]
            {
                "--proxy-server='direct://'",
                "--proxy-bypass-list=*",
                "--disable-infobars",
                "--ignore-certifcate-errors",
                "--no-sandbox"
            }
        });
        await using Page startPage = await browser.NewPageAsync();
        await SetCookies(startPage, "cookies.json");
        Response response = await startPage.GoToAsync(_startUrl);
        
        string[] urls = { ... 3 urls ... }
        Task[] pageTasks = new Task[urls.Length];
        for (int i = 0; i < urls.Length; i++)
        {
            string url = urls[i];
            pageTasks[i] = Task.Run(async () => await GetValuesFromPage(browser, url));
        }
        await Task.WhenAll(pageTasks);

        ...
}

private async Task<PageType> GetValuesFromPage(Browser browser, string url, int layer = 0, bool layerPage = false)
{
        Stopwatch sw = new();
        sw.Start();
        await using Page page = await browser.NewPageAsync();
        await page.SetUserAgentAsync("Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.100 Safari/537.36");
        await page.GoToAsync(url);

       ...

        Task<PageType>[] otherLayersTasks = null;
        if (...) // Other layers exist
        {
            otherLayersTasks = new Task<PageType>[layerSwitchersH.Length];
            for (int i = 0; i < layerSwitchersH.Length; i++)
            {
                int switchToLayer = i;
                otherLayersTasks[i] = Task.Run(async () =>
                    await GetValuesFromPage(browser, url, switchToLayer + 1, true));
            }
        }

       ...
        if (otherLayersTasks != null)
        {
                List<Task> pageTasks = new();
                pageTasks.Add(Task.Run(async () => {
                        PageType[] pts = await Task.WhenAll(otherLayersTasks);
                        ...
                    }));
                await Task.WhenAll(pageTasks);
        }
        sw.Stop();
        Console.WriteLine($"{name}({layer}) is done. {sw.ElapsedMilliseconds}");
}
```

**Comment by kblok** [comment]
Asynchronism does not mean Parallelism. Puppeteer is Async, but we have only one communication channel (a pipe) with the browsers. We can only send and receive one message at a time.
Puppeteer doesn't block the thread while we wait for an answer from the browser, which is the goal of async code.
You can run code in parallel, e.g. While you wait for one tab to finish the navigation you can send a message to open a new tab. But the pipe is only one.

**Comment by 8m0** [comment]
> Asynchronism does not mean Parallelism. Puppeteer is Async, but we have only one communication channel (a pipe) with the browsers. We can only send and receive one message at a time. Puppeteer doesn't block the thread while we wait for an answer from the browser, which is the goal of async code. You can run code in parallel, e.g. While you wait for one tab to finish the navigation you can send a message to open a new tab. But the pipe is only one.

So i can't achieve what i need? O_O

**Comment by kblok** [comment]
Puppeteer-Sharp is using the tools that Chromium provides. You could file an issue [there](https://bugs.chromium.org/p/chromium/issues/list).
